diff -uNr fastfetch-1.9.0.orig/src/fastfetch.c fastfetch-1.9.0/src/fastfetch.c
--- fastfetch-1.9.0.orig/src/fastfetch.c	2023-01-23 13:18:39.637949084 -0600
+++ fastfetch-1.9.0/src/fastfetch.c	2023-01-23 13:19:36.794959308 -0600
@@ -879,6 +879,8 @@
             #define FF_ARCHITECTURE "powerpc"
         #elif defined(__riscv__) || defined(__riscv)
             #define FF_ARCHITECTURE "riscv"
+        #elif defined(__s390x__)
+            #define FF_ARCHITECTURE "s390x"
         #elif
             #define FF_ARCHITECTURE "unknown"
         #endif
